insights/core/plugins.py

```diff
@@ -190,6 +192,28 @@ def register_cluster_reducer(func):
     p["module"] = func.__module__
 
 
+def deprecated(reason='', pending=True):
+    """
+    Mark a parser or combiner as deprecated.  This will emit a warning in
+    the logs when the function is used.
+    """
+    warning_exception = PendingDeprecationWarning if pending else DeprecationWarning
+    # If running under py.test, DeprecationWarnings become errors.
+    if 'py.test' in sys.argv[0]:
```

Comment:
I think using conftest.py to modify state would be preferable.  Won't the `"Warnings enabled"` message print once per usage this way?

Reply:
To be more clear, if in conftest.py you simply `import warnings` and call `warnings.simplefilter()` that will flip the bits necessary when py.test runs.

Reply:
Got it, perfect.

Mind you, we now have this to deal with:
```
=========================== short test summary info ============================
FAIL insights/parsers/tests/test_grub_conf.py::test_grub_conf_raise
FAIL insights/parsers/tests/test_installed_rpms.py::test_release_compare

=================================== FAILURES ===================================
_____________________________ test_grub_conf_raise _____________________________

    def test_grub_conf_raise():
        with pytest.raises(Exception) as e:
            Grub2Config(context_wrap(GRUB2_CFG_4))
>       assert e.value.message == "Cannot parse menuentry line: menuentry {"
E       DeprecationWarning: BaseException.message has been deprecated as of Python 2.6

insights/parsers/tests/test_grub_conf.py:204: DeprecationWarning
_____________________________ test_release_compare _____________________________

    def test_release_compare():
        rpm1 = InstalledRpm.from_package('kernel-rt-debug-3.10.0-327.rt56.204.el7_2.1')
        rpm2 = InstalledRpm.from_package('kernel-rt-debug-3.10.0-327.rt56.204.el7_2.2')
        rpm3 = InstalledRpm.from_package('kernel-3.10.0-327.10.1.el7')
        rpm4 = InstalledRpm.from_package('kernel-3.10.0-327.el7')
        rpm5 = InstalledRpm.from_package('kernel-3.10.0-327.el7_1')
        rpm6 = InstalledRpm.from_package('kernel-3.10.0-327')
        rpm7 = InstalledRpm.from_package('kernel-3.10.0-327.1')
        rpm8 = InstalledRpm.from_package('kernel-3.10.0-327.x86_64')
        rpm9 = InstalledRpm.from_package('kernel-3.10.0-327.1.x86_64')
        with pytest.raises(ValueError) as ve:
            rpm4 < rpm6
>       assert "the other does not" in ve.value.message
E       DeprecationWarning: BaseException.message has been deprecated as of Python 2.6

insights/parsers/tests/test_installed_rpms.py:196: DeprecationWarning
===================== 2 failed, 565 passed in 5.75 seconds =====================
```

Reply:
So I've also fixed those two deprecations.  Hope that's OK :-)